### DISCUSSION TITLE
How to use Recognition Only in Paddle OCR

### DISCUSSION THREAD

**Discussion by ahmadmustafaanis** [PaddlePaddle/PaddleOCR#7600] [Q&A] [answered]
Hi, I do not want my OCR to do detection, I want it to do recognition only, but it only returns a single character.

How can I only run recognizer only, and not detector.

**Comment by matteoPiccoli** [comment, marked as answer]
@ahmadmustafaanis 

First initialize your model with:

`Paddle = paddleocr.PaddleOCR(use_angle_cls=True, lang=your_language, use_gpu=if_gpu_available)`

Then call the recognition only by setting the flags as follows:

`result = Paddle.ocr(image, cls=False, det=False, rec=True)`

In the `result` variable you can find all your text OCR predictions.

**Comment by ahmadmustafaanis** [comment]
@matteoPiccoli 


Here are my results when I am using Det + Rec
![image](https://user-images.githubusercontent.com/47111429/190573078-2fcbe615-5863-4f50-bb8b-58b91814d33c.png)

Here are my results when I am using Rec only on the same image.
![image](https://user-images.githubusercontent.com/47111429/190573165-6c3a6f94-091a-4f22-8853-0b51a22facc1.png)

My ocr initialization is
```
ocr = PaddleOCR(
    use_angle_cls=False,
    lang='en', 
    table=False, 
    use_mp=True,
    # image_dir='ocr_images',
    enable_mkldnn=True,
    use_gpu=True,
    max_batch_size = 20,
    total_process_num = os.cpu_count() * 2 - 1,
    )
```

**Reply by matteoPiccoli** [reply]
@ahmadmustafaanis 

Right now I'm doing a similar thing in my project. Try to detect the text first and then apply the recognition only on the cropped regions of the image contained in the predicted bounding boxes.

Adapt the following pseudo-code according to your needs:

```
import cv2
from paddleocr import PaddleOCR

Paddle = PaddleOCR(use_angle_cls=True, lang="your_language", use_gpu=True)
result = Paddle.ocr(img_path, cls=False, det=True, rec=False)

img= cv2.imread(img_path)
boxes = []
for line in result:
        boxes.append([[int(line[0][0]), int(line[0][1])], [int(line[2][0]), int(line[2][1])]])

for box in boxes:
        cropped_image = copy[box[0][1]:box[1][1], box[0][0]:box[1][0]]
        rec_result = Paddle.ocr(cropped_image, cls=False, det=False, rec=True)
        text, _ = rec_result[0]
```

**Reply by noctisnyxx** [reply]
can we just disable the ppocr DEBUG printout on terminal?

**Reply by Lahari-77** [reply]
yes just set show_log parameter to False during initialisation of paddleocr.
